Content.Client/SimpleStation14/Examine/CharacterInformation/Systems/CharacterInformationSystem.cs

```diff
@@ -0,0 +1,149 @@
+using System.Linq;
+using Content.Client.Examine;
+using Content.Client.Inventory;
+using Content.Shared.SimpleStation14.Examine.CharacterInformation.Components;
+using Content.Client.SimpleStation14.Examine.CharacterInformation.UI;
+using Content.Shared.Access.Components;
+using Content.Shared.CCVar;
+using Content.Shared.DetailExaminable;
+using Content.Shared.PDA;
+using Content.Shared.Roles;
+using Content.Shared.Verbs;
+using Robust.Shared.Configuration;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Client.SimpleStation14.Examine.CharacterInformation.Systems;
+
+public sealed class CharacterInformationSystem : EntitySystem
+{
+    [Dependency] private readonly ExamineSystem _examine = default!;
+    [Dependency] private readonly ClientInventorySystem _inventory = default!;
+    [Dependency] private readonly IEntityManager _entity = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly IConfigurationManager _config = default!;
+
+    private CharacterInformationWindow? _window;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _window = new CharacterInformationWindow();
+
+        SubscribeLocalEvent<CharacterInformationComponent, GetVerbsEvent<ExamineVerb>>(OnGetExamineVerbs);
+    }
+
+
+    private void OnGetExamineVerbs(EntityUid uid, CharacterInformationComponent component, GetVerbsEvent<ExamineVerb> args)
+    {
+        var verb = new ExamineVerb()
+        {
+            Act = () =>
+            {
+                // TODO: Better name?
+                Do(args.Target);
+            },
+            Text = Loc.GetString("character-information-verb-text"),
+            Message = Loc.GetString("character-information-verb-message"),
+            Category = VerbCategory.Examine,
+            Disabled = !_examine.IsInDetailsRange(args.User, uid),
+            Icon = new SpriteSpecifier.Texture(new ResourcePath("/Textures/Interface/VerbIcons/sentient.svg.192dpi.png")),
+            ClientExclusive = true,
+        };
+
+        args.Verbs.Add(verb);
+    }
+
+
+    private void Do(EntityUid uid)
+    {
+        if (_window == null)
+            return;
+
+        string? name = null;
+        string? job = null;
+        string? flavorText = null;
+
+        // Get ID from inventory, get name and job from ID
+        if (_inventory.TryGetSlotEntity(uid, "id", out var idUid))
+        {
+            var id = GetId(idUid);
+            if (id is not null)
+            {
+                var info = GetNameAndJob(id);
+
+                name = info.Item1;
+                job = info.Item2;
+            }
+        }
```

Comment:
Shouldn't this entire thing simply be the GetNameAndJob method? Seems odd to only add a couple lines to the method instead of making it all the logic needed to get the info.